### PR TITLE
[14.0][OU-UPD] rerun analysis for noupdate_changes.xml

### PR DESCRIPTION
**Pull request by remi-filament**
As per #4356 multi-company security rules have changed.
This PR is the outcome of rerunning analysis, then checking manually each file to keep only what is relevant (and not overwriting noupdate_changes.xml which were manually modified while writing migration scripts)

### CODE REVIEW

**Review by legalsylvain — APPROVED** [review]
Thanks !

File: openupgrade_scripts/scripts/crm/14.0.1.2/noupdate_changes.xml [hunk 1]
```diff
@@ -1,8 +1,12 @@
 <?xml version='1.0' encoding='utf-8'?>
 <odoo>
-  <!-- <record id="crm_activity_report_rule_multi_company" model="ir.rule">
+  <record id="crm_activity_report_rule_multi_company" model="ir.rule">
+    <field name="domain_force">[('company_id', 'in', company_ids + [False])]</field>
     <field name="global"/>
```
Comment (MiquelRForgeFlow) [inline openupgrade_scripts/scripts/crm/14.0.1.2/noupdate_changes.xml]: comment this line

File: openupgrade_scripts/scripts/product/14.0.1.2/noupdate_changes.xml [hunk 2]
```diff
@@ -4,18 +4,23 @@
     <!-- <field name="digits" eval="2"/> -->
   </record>
   <record id="product_comp_rule" model="ir.rule">
-<!--    <field name="global"/>-->
+    <field name="domain_force"> [('company_id', 'in', company_ids + [False])]</field>
+    <field name="global"/>
```
Comment (MiquelRForgeFlow) [inline openupgrade_scripts/scripts/product/14.0.1.2/noupdate_changes.xml]: comment global in this file

File: openupgrade_scripts/scripts/sale/14.0.1.1/noupdate_changes.xml [hunk 3]
```diff
@@ -146,10 +148,4 @@
     <field name="name">Personal Invoices</field>
     <field name="domain_force">[('move_type', 'in', ('out_invoice', 'out_refund')), '|', ('invoice_user_id', '=', user.id), ('invoice_user_id', '=', False)]</field>
   </record>
-  <record id="account_invoice_line_rule_see_all" model="ir.rule">
-    <field name="domain_force">[('move_id.move_type', 'in', ('out_invoice', 'out_refund'))]</field>
-  </record>
-  <record id="account_invoice_line_rule_see_personal" model="ir.rule">
-    <field name="domain_force">[('move_id.move_type', 'in', ('out_invoice', 'out_refund')), '|', ('move_id.invoice_user_id', '=', user.id), ('move_id.invoice_user_id', '=', False)]</field>
-  </record>
```
Comment (MiquelRForgeFlow) [inline openupgrade_scripts/scripts/sale/14.0.1.1/noupdate_changes.xml]: don't remove this due to https://github.com/OCA/OpenUpgrade/commit/17e556d243fd4f7bb8960105ab1c884fbf16c160

**Comment by remi-filament** [comment]
Thank @MiquelRForgeFlow for review, I indeed missed a few points ! All resolved

**Review by MiquelRForgeFlow — APPROVED** [review]
LGTM

**Comment by pedrobaeza** [comment]
Thanks for this, Rémi. Remember that `UPD` is not a valid tag. It should be `IMP` or `FIX`. I think that something similar should be done for v15, as there will be some new company rules there. And maybe it's interesting to force the rewriting unconditionally for those coming from a v14 with old rules.

**Comment by remi-filament** [comment]
> And maybe it's interesting to force the rewriting unconditionally for those coming from a v14 with old rules.

I tried asking Odoo to get update script directly in v14 core, but they answered that there is no need, they would get it when upgrading to next version...
I am not sure what would be the best way-forward for existing DB in v14+ : 
1. should we enforce in OpenUpgrade in v15, v16, v17 these noupdate changes for all ir_rules ?
2. should we create a script that could be run directly in existing DB outside of a migration and that could be called by OpenUpgrade (or listed in the doc as prerequisite) when upgrading to v15 / v16 / v17 ?

Solution 1 is probably cleaner, but solution 2 would allow to get performance improvements for existing DB without waiting for migration... (also I am not sure how to make solution 2 since we would need to check for each module if module is installed before updating rules)

**Comment by pedrobaeza** [comment]
FYI, these code snippets serves for updating existing old instances with the new record rules:

v14: https://gist.github.com/pedrobaeza/81047d9d529eae6fc64dff277f68c291
v15: https://gist.github.com/pedrobaeza/4030e735757571632ae457ed6001d6c5
v16: https://gist.github.com/pedrobaeza/c67693945a4e50607670541cee5cfd6c
v17: https://gist.github.com/pedrobaeza/2ce14433272be5a00d89adaab0f5545d

**Pull request merged**